src/host/CommandListPopup.cpp

```diff
@@ -45,15 +45,15 @@ static COORD calculatePopupSize(const CommandHistory& history)
     }
 
     // calculate height, it can range up to 20 rows
-    short height = std::clamp(gsl::narrow<short>(history.GetNumberOfCommands()), minSize.Y, 20i16);
+    short height = std::clamp(gsl::narrow<short>(history->GetNumberOfCommands()), minSize.Y, 20i16);
 
     return { gsl::narrow<short>(width), height };
 }
 
-CommandListPopup::CommandListPopup(SCREEN_INFORMATION& screenInfo, const CommandHistory& history) :
+CommandListPopup::CommandListPopup(SCREEN_INFORMATION& screenInfo, const std::shared_ptr<CommandHistory>& history) :

```

Comment:
The goal is to encapsulate the raw pointer as a smart pointer. You don't have some many choice, it's unique_ptr, weak_ptr or shared_ptr. Ok you can get back a raw pointer but only in small scenarios. I read the guidance and I make the changes. Thanks.

Reply:
I have applied the guidance. It's cool.

Reply:
@ChristophePichaud The CommandListPopup ctor in this case is taking the shared pointer by const reference.   This indicates that this class is not taking part in the ownership.  ie - ref counting is not being incremented.   So this code does not need to use a shared_ptr.  

This code can be left as it was. (taking a const CommandHistory&)   The benefit is you are not tying the pointer type to the class.  If in the future this code changes and uses a different kind of smart pointer this code would not have to change.   

If CommandListPopup really was needed to be part of the ownership chain then this code wouldn't work anyway because it doesn't increment the ref count. 

Does this make sense?

Reply:
> The CommandListPopup ctor in this case is taking the shared pointer by const reference.

 This should be 
The CommandListPopup ctor in this case is taking the shared pointer by const reference and the class is storing the variable in a const & in the class.

Reply:
The CommandListPopup class accepts the history argument two times. First, its stores it as a shared_ptr for future use and also as a const ref for a simple calculation for the bqse ctor... So the code is correct.
Check twice.
Tell me back ?

Reply:
I'm with @dlong11 on this, for this class it needs to have a `CommandHistory` and it can't be changed after a `CommandHistory` class is created. The history also shouldn't be destructed while a popup is around so keeping it a `CommandHistory&` seems more appropriate.

Reply:
Hello @dlong11 and @adiviness, I have made the various changes in the code.
You are right, sometimes, it's better to handle the stuff with references.
Thanks.